File: CrudApi/Entities/Game.cs
using CrudApi.Interfaces;

namespace CrudApi.Entities;

public class Game : ITimestamp, ISoftDelete
{
    public Guid Id { get; set; }
    public required string Title { get; set; }
    public required int ReleaseYear { get; set; }
    public Guid? GenreId { get; set; }
    public Genre? Genre { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.Now;
    public DateTime? UpdatedAt { get; set; }
    public bool IsDeleted { get; set; } = false;
    public DateTime? DeletedAt { get; set; }
}

File: CrudWeb/src/types/game.ts
import type { GenreType } from "./genre";

export interface GameType {
  id: string;
  title: string;
  releaseYear: number;
  genre: GenreType | null;
}

export interface GameCreateType {
  title: string;
  releaseYear: number;
  genreId: string | null;
}

export interface GameUpdateType extends GameCreateType {
  id: string;
}
